apps/studio/src/features/editing-experience/utils/getDgsIdFromString.ts

```diff
@@ -0,0 +1,52 @@
+// DGS Dataset ID utilities for studio form builder
+import { z } from "zod"
+
+const DgsDatasetIdSchema = z.string().regex(/^d_[a-zA-Z0-9]+$/, {
+  message:
+    "DGS dataset ID must start with 'd_' followed by alphanumeric characters",
+})
+
+interface GetDgsIdFromStringProps {
+  string: string
+}
+
+export const getDgsIdFromString = ({
+  string,
+}: GetDgsIdFromStringProps): string | null => {
+  try {
+    // Handle direct ID format: d_abc123
+    if (DgsDatasetIdSchema.safeParse(string).success) {
+      return string
+    }
+
+    // Try to parse as URL
+    const parsedUrl = new URL(string)
+    if (parsedUrl.hostname !== "data.gov.sg") {
+      return null
+    }
```

Comment:
question(non-blocking): for completeness, do we need to support `www.data.gov.sg` also?